include/oneapi/dpl/pstl/hetero/dpcpp/sycl_defs.h

```diff
@@ -21,7 +21,7 @@
 #ifndef _ONEDPL_sycl_defs_H
 #define _ONEDPL_sycl_defs_H
 
-#include <CL/sycl.hpp>
+#include <oneapi/dpl/sycl.hpp>
```

Comment:
We certainly do not want to add a header that can be perceived as a part of public library API set. 

Moreover, I do not think such a header is needed at all. In the single place in the implementation (i.e. this commented line), we can just insert the `__has_include` check etc. directly. For tests, I would find some common utility header (e.g. `utils_sycl.h`) and either also included the SYCL header only there (i.e. also a single place) or defined a macro like this:
```
#if __has_include(<sycl/sycl.hpp>)
#define THE_SYCL_HPP <sycl/sycl.hpp>
#else
#define THE_SYCL_HPP <CL/sycl.hpp>
#endif
```
which could then be used like `#include "utils_sycl.h" #include THE_SYCL_HPP` (newline omitted) in the tests.